File: Trabajo_7/JAVA/Ejercicios/Ejercicio_2.java
package Trabajo_7.Ejercicios;

class Persona {
    protected String correo;
    private String nombre;
    private int edad;
    private float salario;

    public Persona(String nombre, int edad, float salario, String correo) {
        this.nombre = nombre;
        this.edad = edad;
        this.salario = salario;
        this.correo = correo;
    }

    public String getNombre() {return nombre;}

    public int getEdad() {return edad;}

    public float getSalario() {return salario;}

    public String getCorreo() {return correo;}

    public void setCorreo(String correo) {this.correo = correo;}
}

class Empleado extends Persona {
    private int codigo;

    public Empleado(String nombre, int edad, float salario, String correo, int codigo) {
        super(nombre, edad, salario, correo);
        this.codigo = codigo;
    }

    @Override
    public String toString() {
        return "Datos del empleado:\n" +
                "Codigo  -> " + codigo + "\n" +
                "Nombre  -> " + getNombre() + "\n" +
                "Edad    -> " + getEdad() + "\n" +
                "Salario -> " + getSalario() + "\n" +
                "Correo  -> " + getCorreo();
    }
}

class Cliente extends Persona {
    private String estadoCivil;

    public Cliente(String nombre, int edad, float salario, String correo, String estadoCivil) {
        super(nombre, edad, salario, correo);
        this.estadoCivil = estadoCivil;
    }

    @Override
    public String toString() {
        return "Datos del cliente:\n" +
                "Estado  -> " + estadoCivil + "\n" +
                "Nombre  -> " + getNombre() + "\n" +
                "Edad    -> " + getEdad() + "\n" +
                "Salario -> " + getSalario() + "\n" +
                "Correo  -> " + getCorreo();
    }
}

public class Ejercicio_2 {
    public static void main(String[] args) {
        Empleado jose = new Empleado("Jose", 18, 950, "jose.luis@gmail.com", 123);
        Cliente pepe = new Cliente("Pepe", 19, 1000, "pepe.zoto@gmail.com", "Casado");
        jose.setCorreo("Joaquin.data.52@gmail.com");
        System.out.println(jose);
        System.out.println();
        System.out.println(pepe);
    }
}

File: Trabajo_7/CPP/Ejercicios/Ejercicio_2.cpp
#include <iostream>
using namespace std;

class Persona{
protected:
    string correo;

private:
    string nombre;
    int edad;
    float salario;

public:
    Persona(string name, int age, float salary, string mail){
        this -> nombre = name;
        this -> edad = age;
        this -> salario = salary;
        this -> correo = mail;
    }

    //Metodos G y S
    string getNombre() {
        return nombre;
    }

    int getEdad() {
        return edad;
    }

    float getSalario() {
        return salario;
    }

    string &getCorreo(){
        return correo;
    }

};

class Empleado : protected Persona{
private:
    int codigo;
public:
    Empleado(string name, int age, float salary, string mail, int code)
            : Persona(name,age,salary,mail) {
        this -> codigo = code;
    }

    void setCorreo(string mail){
        this->getCorreo() = mail;
    }

    friend ostream& operator << (ostream& os, Empleado Emp_01){
        os << "Datos del empleado:" << endl;
        os << "Codigo  -> " << Emp_01.codigo << endl;
        os << "Nombre  -> " << Emp_01.getNombre()<< endl;
        os << "Edad    -> " << Emp_01.getEdad() << endl;
        os << "Salario -> " << Emp_01.getSalario() << endl;
        os << "Correo  -> " << Emp_01.getCorreo() << endl;

        return os;
    }



};

class Cliente : protected Persona{
private:
    string estado_civil;
public:
    Cliente(string name, int age, float salary, string mail, string civil_status)
            : Persona(name,age,salary,mail) {
        this -> estado_civil = civil_status;
    }

    void setCorreo(string mail){
        this->getCorreo() = mail;
    }

    friend ostream& operator << (ostream& os, Cliente Cli_01){
        os << "Datos del cliente:" << endl;
        os << "Estado  -> " << Cli_01.estado_civil << endl;
        os << "Nombre  -> " << Cli_01.getNombre()<< endl;
        os << "Edad    -> " << Cli_01.getEdad() << endl;
        os << "Salario -> " << Cli_01.getSalario() << endl;
        os << "Correo  -> " << Cli_01.getCorreo() << endl;

        return os;
    }

};


int main(){
    Empleado Jose("Jose", 18, 950, "jose.luis@gmail.com", 123);
    Cliente Pepe("Pepe", 19, 1000, "pepe.zoto@gmail.com", "Casado");
    Jose.setCorreo("Joaquin.data.52@gmail.com");
    cout << Jose;
    cout << "" << endl;
    cout << Pepe;

    return 0;
}